src/rpc/mod.rs

```diff
@@ -181,17 +181,23 @@ impl ConnectionSender {
     }
 
     #[allow(dead_code)]
-    /// close the connection
+    /// Close the connection
+    /// Requests in requests map will be cancelled when the writer closed
     async fn close(&mut self) -> Result<()> {
         self.request(ObTablePacket::ClosePoison).await?;
-        self.shutdown()
+        if let Some(writer) = mem::take(&mut self.writer) {
+            writer.await??
+        }
+        self.sender.closed().await;
```

Comment:
I guess there is no need to check wait for `sender` to be closed? What we cares about is whether the tcp send worker is stopped.

Reply:
Fixed.